File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        unordered_map<int,int> lastIndex; 
        for (int i = 0; i < (int)nums.size(); ++i) {
            auto it = lastIndex.find(nums[i]);
            if (it != lastIndex.end()) {
                if (i - it->second <= k) return true;
            }
            lastIndex[nums[i]] = i;
        }
        return false;
    }
};

File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        d = {}
        for i in range(len(nums)):
            if nums[i] not in d:
                d[nums[i]] = i
            else:
                prevIdx = d.get(nums[i])
                if(abs(prevIdx - i)<=k):
                    return True
                else:
                    d[nums[i]] = i
        return False